hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/BaseTableServicePlanActionExecutor.java

```diff
@@ -124,7 +124,9 @@ public Pair<Option<HoodieInstant>, Set<String>> getIncrementalPartitions(TableSe
         .filter(this::filterCommitByTableType).flatMap(instant -> {
           try {
             String completionTime = instant.getCompletionTime();
-            if (completionTime.compareTo(leftBoundary) >= 0 && completionTime.compareTo(rightBoundary) < 0) {
+            if (completionTime.compareTo(leftBoundary) >= 0
+                && ((instant.requestedTime().length() < completionTime.length() && instant.requestedTime().compareTo(rightBoundary) < 0)
```

Comment:
can we add java docs please.

Reply:
Reverted this change. This was added only for running some tests.